### PR TITLE
JAPI-376 Introduce forced timeout junit tests for DFUClient reads/writes

### PR DESCRIPTION
**Pull request by jpmcmu**
Signed-off-by: James McMullan James.McMullan@lexisnexis.com

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [X] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [X] I have created a corresponding JIRA ticket for this submission
- [X] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [X] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [X] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [X] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by jpmcmu** [comment]
@rpastrana please review

File: dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSReadWriteTest.java [hunk 1]
```diff
@@ -67,7 +67,16 @@ public void readBadlyDistributedFileTest() throws Exception
...
+        List<HPCCRecord> records = readFile(file, connTO, false);
+        assertEquals("Not all records loaded",expectedCounts[1], records.size());
+    }
+
+    @Test
+    public void readWithForcedTimeoutTest() throws Exception
+    {
+        HPCCFile file = new HPCCFile(datasets[0], connString , hpccUser, hpccPass);
+
+        List<HPCCRecord> records = readFile(file, connTO, true);
```
Comment (rpastrana) [inline dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSReadWriteTest.java:79]: As far as I can see, connTO will be null here unless it is overwritten via system property, and getFileAccessExpirySecs is then defaulted in dfsclient.hpccfile to 120 secs, which seems like an unnecessary wait for forced timeout test. Why don't we lower the connTO for this test to 1 or 2 secs?

**Review by rpastrana — COMMENTED** [review]
@jpmcmu looks good, just a comment on the actual timeout value for the forced timeout test.

**Comment by rpastrana** [comment]
@jpmcmu github action tests reported this issue:
 Error:  Failures: 
Error:    DFSReadWriteTest.readWithForcedTimeoutTest:80 Not all records loaded expected:<5600> but was:<1250>

**Comment by jpmcmu** [comment]
@rpastrana changed the timeout and fixed the record count discrepancy issue

**Review by rpastrana — APPROVED** [review]
@jpmcmu looks good

**Pull request merged**